test/inference-engines/threeCliques.test.ts

```diff
@@ -1,207 +1,49 @@
-
-  expect(infer({ A: ['T'] }).toFixed(4)).toBe('0.5000')
-  expect(infer({ A: ['F'] }).toFixed(4)).toBe('0.5000')
-  expect(infer({ B: ['T'] }).toFixed(4)).toBe('0.5000')
-  expect(infer({ B: ['F'] }).toFixed(4)).toBe('0.5000')
-  expect(infer({ C: ['T'] }).toFixed(4)).toBe('0.1500')
-  expect(infer({ C: ['F'] }).toFixed(4)).toBe('0.8500')
-  expect(infer({ D: ['T'] }).toFixed(4)).toBe('0.5000')
-  expect(infer({ D: ['F'] }).toFixed(4)).toBe('0.5000')
-  expect(infer({ E: ['T'] }).toFixed(4)).toBe('1.0000')
-  expect(infer({ E: ['F'] }).toFixed(4)).toBe('0.0000')
-}
-
-const tests: { [key: string]: (engine: IInferenceEngine) => void } = {
-  'infers give nothing': infersGiveNothing,
-  'infers give A False': infersGiveAFalse,
-  'infers give A True': infersGiveATrue,
-  'infers give B False': infersGiveBFalse,
-  'infers give B True': infersGiveBTrue,
-  'infers give C False': infersGiveCFalse,
-  'infers give C True': infersGiveCTrue,
-  'infers give D False': infersGiveDFalse,
-  'infers give D True': infersGiveDTrue,
-  'infers give E False': infersGiveEFalse,
-  'infers give E True': infersGiveETrue,
-}
-
-describe('infers', () => {
-  describe('3q network', () => {
-    const testNames = Object.keys(tests)
-    const engine = new InferenceEngine(network)
-
-    for (const testName of testNames) {
-      const method = tests[testName]
-      it(`${testName}`, () => method(engine))
-    }
-  })
-})
+import { runTests } from './helpers'
+
+// These gold standard values were computed indepdendently in R.
+const GOLD_STANDARD: [string[], number][] = [
+  [['T', 'T', 'T', 'T', 'T'], 0.00625],
+  [['F', 'T', 'T', 'T', 'T'], 0.00625],
+  [['T', 'F', 'T', 'T', 'T'], 0.0125],
+  [['F', 'F', 'T', 'T', 'T'], 0.0125],
+  [['T', 'T', 'F', 'T', 'T'], 0.05625],
+  [['F', 'T', 'F', 'T', 'T'], 0.05625],
+  [['T', 'F', 'F', 'T', 'T'], 0.05],
+  [['F', 'F', 'F', 'T', 'T'], 0.05],
+  [['T', 'T', 'T', 'F', 'T'], 0.00625],
+  [['F', 'T', 'T', 'F', 'T'], 0.00625],
+  [['T', 'F', 'T', 'F', 'T'], 0.0125],
+  [['F', 'F', 'T', 'F', 'T'], 0.0125],
+  [['T', 'T', 'F', 'F', 'T'], 0.05625],
+  [['F', 'T', 'F', 'F', 'T'], 0.05625],
+  [['T', 'F', 'F', 'F', 'T'], 0.05],
+  [['F', 'F', 'F', 'F', 'T'], 0.05],
+  [['T', 'T', 'T', 'T', 'F'], 0.00625],
+  [['F', 'T', 'T', 'T', 'F'], 0.00625],
+  [['T', 'F', 'T', 'T', 'F'], 0.0125],
+  [['F', 'F', 'T', 'T', 'F'], 0.0125],
+  [['T', 'T', 'F', 'T', 'F'], 0.05625],
+  [['F', 'T', 'F', 'T', 'F'], 0.05625],
+  [['T', 'F', 'F', 'T', 'F'], 0.05],
+  [['F', 'F', 'F', 'T', 'F'], 0.05],
+  [['T', 'T', 'T', 'F', 'F'], 0.00625],
+  [['F', 'T', 'T', 'F', 'F'], 0.00625],
+  [['T', 'F', 'T', 'F', 'F'], 0.0125],
+  [['F', 'F', 'T', 'F', 'F'], 0.0125],
+  [['T', 'T', 'F', 'F', 'F'], 0.05625],
+  [['F', 'T', 'F', 'F', 'F'], 0.05625],
+  [['T', 'F', 'F', 'F', 'F'], 0.05],
+  [['F', 'F', 'F', 'F', 'F'], 0.05],
+]
+
+const names: string[] = ['A', 'B', 'C', 'D', 'E']
+const testValues: string[][][] = [
+  [['T'], ['F'], ['T', 'F']],
```

Comment:
Are all of these lines supposed to be the same?

Reply:
Yes.   Each variable has the levels "T" and "F".   This will test all possible combinations where the event contains no value for the variable, a single value, either "T" or "F", or both values.